src/fu-engine.c

```diff
@@ -5953,20 +5960,37 @@ fu_engine_ensure_security_attrs(FuEngine *self)
 
 	/* distil into one simple string */
 	g_free(self->host_security_id);
-	self->host_security_id = fu_engine_attrs_calculate_hsi_for_chassis(self);
+	self->host_security_id = fu_engine_attrs_calculate_hsi_for_chassis(self, &hsi_number);
+
+	if (fu_history_get_last_hsi_details(self->history,
+					    &previous_hsi,
+					    &last_json_attr,
+					    &error)) {
```

Comment:
I think it's better to reverse this; it's more common to see `if (!fu_history_get_last_hsi_details` e.g. where you handle the failure case first.

Reply:
Ok. Let me double-check the code and handle the failure case first.